port/raspberrypi/rp2xxx/src/hal/drivers.zig

```diff
@@ -73,11 +73,20 @@ pub const I2C_Device = struct {
         return microzig.utilities.Slice_Vector([]u8).init(datagrams).size();
     }
 
+    pub fn write_then_read(dev: I2C_Device, src: []const u8, dst: []u8) !void {
+        try dev.bus.write_then_read_blocking(dev.address, src, dst, null);
+    }
+
+    pub fn write_then_readv(dev: I2C_Device, datagrams: []const []const u8) !void {
+        try dev.bus.writev_blocking(dev.address, datagrams, null);
+    }
+
     const vtable = Datagram_Device.VTable{
         .connect_fn = null,
         .disconnect_fn = null,
         .writev_fn = writev_fn,
         .readv_fn = readv_fn,
+        .write_then_readv_fn = write_then_readv_fn,
```

Comment:
You have to add this (just set to null) to all the other drivers otherwise they won't compile